src/ConnectionPool/SentinelConnectionPool.php

```diff
@@ -0,0 +1,274 @@
+                    !empty($normalizedRepolicaData['ip']) &&
+                    !empty($normalizedRepolicaData['port'])
+                ) {
+                    $replicaKey = $normalizedRepolicaData['ip'] . ':' . $normalizedRepolicaData['port'];
+                    if (isset($this->replicasConnection[$replicaKey])) {
+                        continue;
+                    }
+
+                    $this->replicas[$replicaKey] = [
+                        'ip' => $normalizedRepolicaData['ip'],
+                        'port' => $normalizedRepolicaData['port'],
+                    ];
+                }
+            }
+            $this->failedCount = 0;
+            $sentinelConnection->call('close');
+            return true;
+        }
+
+        return false;
+    }
+
+    private function getMasterConnection()
+    {
+        return $this->masterConnection;
+    }
+
+    /**
+     * @throws RedisProxyException
+     */
+    private function getReplicaConnection()
+    {
+        if (count($this->replicas) > 0) {
+            while ($replica = array_shift($this->replicas)) {
+                try {
+                    $replicaConnection = $this->driver->getConnectionFactory()->create($replica['ip'], $replica['port'], $this->timeout);
+                    $this->driver->connectionSelect($replicaConnection, $this->database);
+
+                    $role = $this->driver->connectionRole($replicaConnection);
+                    if ($role !== 'slave') {
+                        continue;
+                    }
+
+                    $this->replicasConnection[] = $replicaConnection;
+                    return $replicaConnection;
+                } catch (RedisProxyException $e) {
+                    throw $e;
+                } catch (Throwable $t) {
+                    continue;
+                }
+            }
+        }
+
+        if (count($this->replicasConnection) === 0) {
+            return null;
+        }
+
+        return $this->replicasConnection[array_rand($this->replicasConnection)];
+    }
+
+    private function shiftSentinels(): void
+    {
+        $sentinel = array_shift($this->sentinels);
+        $this->sentinels[] = $sentinel;
+    }
+
+    private function reset(): void
+    {
+        $this->masterConnection = null;
+        $this->replicas = [];
+        $this->replicasConnection = [];
+    }
+
+    private function normalizeResponze(array $arr): array
+    {
+        $keys = array_values(array_filter($arr, function ($key) {
+            return $key % 2 == 0;
+        }, ARRAY_FILTER_USE_KEY));
+        $values = array_values(array_filter($arr, function ($key) {
+            return $key % 2 == 1;
+        }, ARRAY_FILTER_USE_KEY));
+
+        if (count($keys) != count($values)) {
+            throw new RedisProxyException("Wrong number of arguments");
+        }
+        return array_combine($keys, $values);
+    }
+    
+    private function getReadOnlyOperations(): array
+    {
+        return [
```

Comment:
tot by som dal asi do nejakej konstanty ci?

Reply:
Ono to moze byt len pre ten sentinel pool, pri master/slave poole by to mohol byt zase iny set. Preto som to chcel drzat v jednej triede a nerobit zbytocne dalsie. A hore do classy som to davat nechcel nech je prehladnejsia.